From 432ba02fe7576f8ce807e8c7f07ed82fbb0727bc Mon Sep 17 00:00:00 2001
From: Martin Perina <mperina@redhat.com>
Date: Tue, 15 Apr 2014 10:50:53 +0200
Subject: [RHEL6.6 PATCH 1/4 v2] Add function is_pcs_fence_kdump

Extracts relevant parts of code into function is_pcs_fence_kdump.
This is part of code cleanup prior to introduction of fence_kdump
configuration for generic clusters.

Related: rhbz#1083938
Signed-off-by: Martin Perina <mperina@redhat.com>
---
 mkdumprd | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/mkdumprd b/mkdumprd
index e0814ad..43c526b 100644
--- a/mkdumprd
+++ b/mkdumprd
@@ -1516,6 +1516,13 @@ kdump_chk()
     cleanup_and_exit 1
 }
 
+# Tests if fence_kdump is configured in Pacemaker cluster.
+is_pcs_fence_kdump()
+{
+    [ -x /usr/sbin/fence_kdump_send ] || return 1
+    [ -f "$CLUSTER_CONFIG_FILE" ] || return 1
+}
+
 if [ -z "$MNTIMAGE" -o -z "$IMAGE" ]; then
     error "Error creating temporaries.  Try again"
     cleanup_and_exit 1
@@ -1770,7 +1777,7 @@ if [ "$DEFAULT_ACTION" == "mount_root_run_init" ];then
 fi
 
 # handle cluster node if CLUSTER_CONFIG_FILE and fence_kudmp_send exist
-if [ -f "$CLUSTER_CONFIG_FILE" -a -e /usr/sbin/fence_kdump_send ]; then
+if is_pcs_fence_kdump; then
     # get cluster nodes from CLUSTER_CONFIG_FILE, get interface and ip addresses
     nodelist=`echo "xpath /cluster/clusternodes/clusternode/@name" | xmllint --shell $CLUSTER_CONFIG_FILE | grep content | cut -d'=' -f2`
     for node in ${nodelist}; do
-- 
1.8.3.1

